extmod/nimble/hal/hal_uart.c

```diff
@@ -86,15 +86,37 @@ int hal_uart_close(uint32_t port) {
     return 0; // success
 }
 
+STATIC void mp_bluetooth_hci_uart_readchar_cb(uint8_t chr) {
```

Comment:
Rename to `mp_bluetooth_hci_uart_char_cb` (I saw "readchar" in the name and assumed it was for the byte-by-byte path only).

Also as this is the common point for both, move the

```
#if HCI_TRACE
        printf("> %02x\n", chr);
        #endif
 ```

here.